soh/soh/OTRGlobals.cpp

```diff
@@ -925,6 +927,84 @@ bool PathTestCleanup(FILE* tfile) {
     return true;
 }
 
+// TODO: Copied from randomzier, find a better place for it
+std::string sanitizePath(std::string stringValue) {
+    // Add backslashes.
+    for (auto i = stringValue.begin();;) {
+        auto const pos = std::find_if(i, stringValue.end(), [](char const c) { return '\\' == c || '\'' == c || '"' == c; });
+        if (pos == stringValue.end()) {
+            break;
+        }
+        i = std::next(stringValue.insert(pos, '\\'), 2);
+    }
+
+    // Removes others.
+    stringValue.erase(std::remove_if(stringValue.begin(), stringValue.end(), [](char const c) {
+        return '\n' == c || '\r' == c || '\0' == c || '\x1A' == c; }), stringValue.end());
+
+    return stringValue;
+}
+
+static const char* lastDroppedFile = "";
+
+void processDroppedFiles() {
+    const char* droppedFile = CVarGetString("gDroppedFile", "");
+    if (strcmp(droppedFile, lastDroppedFile) != 0 && strcmp(droppedFile, "") != 0) {
```

Comment:
i know with rando we use `gNewFileDropped` to not need to do string compares to see if we have a new file, is there a specific reason to not use that here too?

Reply:
Ideally we move away from this system towards hooks or something anyway, I'm mostly just trying to make due with what we have while not touching the other parts, in hopes of not breaking something.

I mostly wasn't sure if I could even rely on `gNewFileDropped` because the file select screen sets it to 0 when it's done with it, and I'm not sure where my process is relative to that.

Reply:
Now using `gNewFileDropped` on the file select screen since I can guarantee the order.